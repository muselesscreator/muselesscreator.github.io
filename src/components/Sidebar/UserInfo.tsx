type UserInfoProps = {
  contactInfo: {
    name: string;
    title: string;
  };
};

const UserInfo = ({ contactInfo }: UserInfoProps) => {
  return (
    <div className="info-content">
      <h1 className="name" title={contactInfo.name}>{contactInfo.name}</h1>
      <p className="title">{contactInfo.title}</p>
    </div>
  );
};

export default UserInfo;
