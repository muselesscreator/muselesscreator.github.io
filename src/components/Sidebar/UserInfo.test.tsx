import {
  describe,
  test,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';
import contactInfo from '~/data/contact';

import UserInfo from './UserInfo';

describe('UserInfo component', () => {
  test('snapshot includes user name and title', () => {
    const el = shallow(<UserInfo />);
    expect(el.snapshot).toMatchSnapshot();
    expect(el.instance.findByType('h1')[0].children[0].el).toEqual(contactInfo.name);
    expect(el.instance.findByType('p')[0].children[0].el).toEqual(contactInfo.title);
  });
});
