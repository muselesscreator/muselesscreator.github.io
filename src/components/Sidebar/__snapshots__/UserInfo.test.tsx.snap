// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserInfo component > snapshot includes user name and title 1`] = `
<div
  className="info-content"
>
  <h1
    className="name"
    title="Ben Warzeski"
  >
    Ben Warzeski
  </h1>
  <p
    className="title"
  >
    Full-Stack Software Engineer
  </p>
</div>
`;
